lib/src/clojure_lsp/feature/file_management.clj

```diff
@@ -84,37 +85,56 @@
-        source-paths (settings/get db [:source-paths])
-        changed-var-definitions (find-changed-var-definitions old-local-analysis new-local-analysis)
-        references-filenames (->> changed-var-definitions
-                                  ;; TODO: switch from nested-loop join to hash join
-                                  (mapcat #(q/find-references project-analysis % false db))
-                                  (map :filename))
+  (let [changed-var-definitions (find-changed-var-definitions old-local-analysis new-local-analysis)
+        ;; TODO: can we easily remove usages of external namespaces (clojure.core, etc.) here?
         changed-var-usages (find-changed-var-usages old-local-analysis new-local-analysis)
-        definitions-filenames (->> changed-var-usages
-                                   ;; TODO: remove usages of external namespaces (clojure.core, etc.) here
-                                   ;; TODO: switch from nested-loop join to hash join
-                                   (keep #(q/find-definition project-analysis % db))
-                                   (filter (fn [d]
-                                             ;; TODO: this excludes "private calls", but they are counted in code lens, so maybe shouldn't exclude
-                                             (and (not (:private d))
-                                                  ;; TODO: is this extra work? The definition came from project-analysis, so should be on the source-paths.
-                                                  (some #(string/starts-with? (:filename d) %) source-paths))))
-                                   (map :filename))]
-    (disj (set (concat references-filenames definitions-filenames))
-          filename)))
+        project-analysis (into {}
+                               (q/filter-project-analysis-xf db)
+                               (dissoc (:analysis @db) filename)) ;; don't notify self
+        incoming-filenames (when (seq changed-var-definitions)
+                             (let [def-signs (->> changed-var-definitions
+                                                  (map q/var-definition-signatures)
+                                                  (apply set/union))]
+                               (into #{}
+                                     (comp
+                                       (mapcat val)
+                                       (filter #(identical? :var-usages (:bucket %)))
+                                       (filter #(contains? def-signs (q/var-usage-signature %)))
+                                       (map :filename))
+                                     project-analysis)))
+        outgoing-filenames (when (seq changed-var-usages)
+                             (let [usage-signs->langs (->> changed-var-usages
+                                                              ;; TODO: do we really care if lang doesn't match?
```

Comment:
For .cljc files, clj-kondo returns 2 analysis for the same element, one for clj and other for cljs, some times one of then have its particularity for the lang. We need to think if that could affect this notify-references feature, I don't see any problems not checking lang here, but maybe I could be missing something.